### DISCUSSION TITLE
Feature Request: Role Mapping for LDAP Integration

### DISCUSSION THREAD

**Discussion by GabrielRamirez** [open-webui/open-webui#7245] [Ideas]
I’d like to suggest a new feature for the LDAP integration: **Role Mapping**.  

This feature would allow roles defined in LDAP/AD (e.g., `Admin`) to be mapped to corresponding roles within Open WebUI. For example:  

- A user in the `Admin` group in LDAP/AD would automatically be assigned the `Admin` role in Open WebUI.  

**Why is this useful?**  

- Simplifies user management by automatically assigning roles based on LDAP/AD settings.  
- Ensures consistent role management across the organization.  
- Reduces administrative overhead by eliminating manual role assignments in Open WebUI.  

**Use Case Example:**  
1. An organization manages users in LDAP/AD with roles such as `Admin`, `Editor`, and `Viewer`.  
2. When a user logs into Open WebUI via LDAP, their role (e.g., `Admin`) is automatically mapped and assigned.  
3. This ensures roles remain consistent with the organization's policies and saves time.

**Comment by skimike02** [comment]
It would be great if it could also update user groups based on LDAP roles.

Here is a thought on how this part could work:

1. In the environment variables:
        - make LDAP_USER_GROUPS_FIELD. This should store the LDAP attribute that stores a list of what groups the user should be a member of. It may also contain groups that are not OWUI groups. These will be ignored.

2. In apps.webui.models.groups:
        - `def group_sync(user_id: str, groups: list[str]): """This function will iteratively update groups to add/remove the userid to ensure they have access to the groups listed and only the groups listed"""`
        - If the user is in an OWUI group not listed in the LDAP field, the user will be removed using update_group_by_id
        - If the user is not in an OWUI group that is listed in the LDAP field, they will be added using update_group_by_id.

3. In apps.webui.routers.auths.py: Add to the `ldap_auth` function:
        - if LDAP_USER_GROUPS_FIELD is set, retrieve the groups from LDAP and run the group_sync function

**Comment by Soap2G** [comment]
Hi, are there any updates on this?
